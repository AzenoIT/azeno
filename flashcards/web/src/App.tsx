import { router } from "router";
import { RouterProvider } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { AuthConfig, AuthTokenProvider } from "modules/common/AuthToken";
import ErrorBoundary from "modules/common/components/ErrorBoundary";
import FormikTest from "modules/genericTests/FormikTest/FormikTest";
import TailwindTest from "modules/genericTests/TailwindTest/TailwindTest";
import FakerTest from "modules/genericTests/FakerTest/FakerTest";
import MaterialUITest from "modules/genericTests/MaterialUITest/MaterialUITest";
import AddDecksForm from "components/AddDecksForm";

function App() {
    const queryClient = new QueryClient();
    const authConfig: AuthConfig = { loginEndpoint: "/api/v1/token/", refreshEndpoint: "/api/v1/token/refresh/" };

    return (
        <QueryClientProvider client={queryClient}>
            <AuthTokenProvider config={authConfig}>
                <ErrorBoundary>
                    <RouterProvider router={router} />
                    <FakerTest />
                    <TailwindTest />
                    <FormikTest />
                    <MaterialUITest />
                    <AddDecksForm />
                </ErrorBoundary>
            </AuthTokenProvider>
        </QueryClientProvider>
    );
}

export default App;
