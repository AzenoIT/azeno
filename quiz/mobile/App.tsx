import { StyleSheet, Text, View, SafeAreaView, StatusBar, TextInput, Button } from "react-native";
import React from "react";
import { Formik } from "formik";
import * as yup from "yup";
// eslint-disable-next-line import/no-extraneous-dependencies
import { NavigationContainer } from "@react-navigation/native";
// eslint-disable-next-line import/no-extraneous-dependencies
import { createStackNavigator } from "@react-navigation/stack";

const loginValidationSchema = yup.object().shape({
    email: yup.string().email("Please enter valid email").required("Email Address is Required"),
});

const styles = StyleSheet.create({
    loginContainer: {
        width: "80%",
        alignItems: "center",
        padding: 10,
        elevation: 10,
        backgroundColor: "#e6e6e6",
    },
    textInput: {
        height: 40,
        width: "100%",
        margin: 10,
        backgroundColor: "white",
        borderColor: "gray",
        borderWidth: StyleSheet.hairlineWidth,
        borderRadius: 10,
    },
});

// @ts-ignore
function HomeScreen({ navigation }) {
    return (
        <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
            <Button title="Go to Login Screen" onPress={() => navigation.navigate("Login")} />
        </View>
    );
}

function Login() {
    return (
        <>
            <StatusBar barStyle="dark-content" />
            <SafeAreaView style={styles.loginContainer}>
                <View>
                    <Text>Login Screen</Text>
                </View>
            </SafeAreaView>
            <Formik validationSchema={loginValidationSchema} initialValues={{ email: "" }} onSubmit={() => {}}>
                {({ handleChange, handleBlur, handleSubmit, values, errors, isValid }) => (
                    <>
                        <TextInput
                            placeholder="Email Address"
                            style={styles.textInput}
                            onChangeText={handleChange("email")}
                            onBlur={handleBlur("email")}
                            value={values.email}
                            keyboardType="email-address"
                        />
                        {errors.email && <Text style={{ fontSize: 10, color: "red" }}>{errors.email}</Text>}
                        <Button onPress={handleSubmit} title="LOGIN" disabled={!isValid} />
                    </>
                )}
            </Formik>
        </>
    );
}

const Stack = createStackNavigator();

function MyStack() {
    return (
        <Stack.Navigator>
            <Stack.Screen name="Home" component={HomeScreen} />
            <Stack.Screen name="Login" component={Login} />
        </Stack.Navigator>
    );
}

export default function App() {
    return (
        <NavigationContainer>
            <MyStack />
        </NavigationContainer>
    );
}
