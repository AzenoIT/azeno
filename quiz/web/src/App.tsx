import Formik from "./components/FormikTest/Formik";

function App() {
    return (
        <div className="App">
            <Formik />
        </div>
    );
}

export default App;
