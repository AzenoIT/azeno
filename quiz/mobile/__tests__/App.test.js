import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";

import App from "../App";

describe("<App />", () => {
    it("has 1 child", () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree.children.length).toBe(1);
    });
});

it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
