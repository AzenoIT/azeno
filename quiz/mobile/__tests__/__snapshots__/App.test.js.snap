// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;
